DataConverter/Catalyst.DataConverter/HierarchicalDataTransformer.cs

```diff
@@ -66,38 +79,361 @@ public HierarchicalDataTransformer(IHierarchicalDataTransformerHelper helper)
+
+            var queryConfig = new QueryConfig
+                                  {
+                                      ConnectionString = deserialzed.ConnectionString,
+                                      Url = deserialzed.Url,
+                                      MaximumEntitiesToLoad = deserialzed.MaximumEntitiesToLoad,
+                                      EntitiesPerBatch = deserialzed.EntitiesPerBatch,
+                                      EntitiesPerUploadFile = deserialzed.EntitiesPerUploadFile,
+                                      LocalSaveFolder = deserialzed.LocalSaveFolder,
+                                      DropAndReloadIndex = deserialzed.DropAndReloadIndex,
+                                      WriteTemporaryFilesToDisk = deserialzed.WriteTemporaryFilesToDisk,
+                                      WriteDetailedTemporaryFilesToDisk = deserialzed.WriteDetailedTemporaryFilesToDisk,
+                                      CompressFiles = deserialzed.CompressFiles,
+                                      UploadToElasticSearch = deserialzed.UploadToElasticSearch,
+                                      Index = deserialzed.Index,
+                                      Alias = deserialzed.Alias,
+                                      EntityType = deserialzed.EntityType,
+                                      TopLevelKeyColumn = deserialzed.TopLevelKeyColumn,
+                                      UseMultipleThreads = deserialzed.UseMultipleThreads,
+                                      KeepTemporaryLookupColumnsInOutput = deserialzed.KeepTemporaryLookupColumnsInOutput,
+                                  };
+
+            return queryConfig;
+        }
+
+        private async Task<JobData> GetJobData(Binding binding, Entity destinationEntity)
+        {
+            var jobData = new JobData();
+
+            Binding[] allBindings = await this.GetBindingsForEntityAsync(destinationEntity);
+
+            this.ValidateHierarchicalBinding(binding, allBindings);
+
+            List<DataSource> dataSources = new List<DataSource>();
+
+            await this.GenerateDataSources(binding, allBindings, destinationEntity, dataSources, null, "$", isFirst: true);
+
+            jobData.MyDataSources = dataSources;
+
+            return jobData;
+        }
+
+        /// <summary>
+        /// Execute DataBus with the given configuration and job data
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="jobData"></param>
+        private void RunDatabus(QueryConfig config, JobData jobData)
+        {
+            LoggingHelper2.Debug(this.guid, "We are trying to run Databus");
+            var job = new Job
+                          {
+                              Config = config,
+                              Data = jobData
+                          };
+            try
+            {
+                var runner = new DatabusRunner();
```

Comment:
do we need a new DatabusRunner every time or could that be new'd up in the constructor?